signer/src/storage/postgres.rs

```diff
@@ -1566,14 +1686,34 @@ impl super::DbRead for PgStore {
     async fn get_signer_utxo(
         &self,
         chain_tip: &model::BitcoinBlockHash,
-        context_window: u16,
     ) -> Result<Option<SignerUtxo>, Error> {
-        let txo_type = model::TxOutputType::SignersOutput;
-        if let Some(pg_utxo) = self.get_utxo(chain_tip, context_window, txo_type).await? {
-            return Ok(Some(pg_utxo));
+        // If we've swept funds before, then will have a signer output, so
+        // let's try that first.
+        let output_type = model::TxOutputType::SignersOutput;
+        if let Some(min_block_height) = self.minimum_utxo_height(output_type).await? {
+            let utxo_fut = self.get_utxo(chain_tip, output_type, min_block_height);
+
+            match utxo_fut.await? {
+                Some(pg_utxo) => return Ok(Some(pg_utxo)),
+                // This means the signer UTXO has been reorged and the only
+                // outputs that we know about are donations.
+                None => {
+                    let output_type = model::TxOutputType::Donation;
+                    let utxo_fut = self.get_utxo(chain_tip, output_type, min_block_height);
+                    return utxo_fut.await;
+                }
+            }
         }
-        let txo_type = model::TxOutputType::Donation;
-        self.get_utxo(chain_tip, context_window, txo_type).await
+
+        // We do not have a signer output. Maybe we have a donation, so
+        // let's try that.
+        let output_type = model::TxOutputType::Donation;
+        let Some(min_block_height) = self.minimum_utxo_height(output_type).await? else {
+            return Ok(None);
+        };
+
+        self.get_utxo(chain_tip, output_type, min_block_height)
+            .await
```

Comment:
The logic for fetching donations needs some work still (it's "wrong" here). But the stuff above should be correct.

Reply:
I think the logic is correct after https://github.com/stacks-network/sbtc/pull/1048/commits/c6c914f39252f59c3c3715f53cabfc5b0ec826dd. Now for more tests

Reply:
I just realized that the above was wrong. Fixed in https://github.com/stacks-network/sbtc/pull/1048/commits/a5fdb3f05665e78e5de1c319fb0ea57464319228... let's hope that this is the last fix.

Also, I cannot help but think that there is a simpler way to get the signers' UTXO. Any ideas?